R/toolISOhistorical.R

```diff
@@ -111,61 +112,293 @@ toolISOhistorical <- function(m, mapping = NULL, additional_mapping = NULL, over
     }
     return(tr)
   }
-
+  
   tr <- .identifyTransitions(m)
   vcat(2, "The following transitions are found in the data \n", paste(tr, collapse = ", \n"))
-
-  # loop over all transitions
-  for (a in tr) {
-    # check if new regions of transition exists in m
-    toISOmissing   <- !all(is.element(a$toISO, getItems(m, dim = 1.1)))
-    if (toISOmissing) stop("there is no data for the following new countrys: ",
-                           paste(a$toISO[which(!is.element(a$toISO, getItems(m, dim = 1.1)))], collapse = ", "))
-
-    # create transformation matrix
-    ## time span where the data need to be adjusted
-    subTime <- getYears(m[, seq_len(which(getYears(m) == a$fromY)), ])
-    # disaggregation of countries
-    if (length(a$fromISO) == 1) {
-
-      if (is.null(additional_weight)) {
-        weight <- setYears(m[a$toISO, a$toY, ], NULL)
-        if (anyNA(weight)) {
-          weight[is.na(weight)] <- 0
-          vcat(0, "Weight in toolISOhistorical contained NAs. Set NAs to 0!")
+  
+  if (ndim(m, dim = 1) == 2) {
```

Comment:
The same condition is checked again in line 178, so this block could be moved there, right?